### PR TITLE
Add land development structure

### PR DESCRIPTION
**Pull request by pmargreff**
First version for land development for early feedback in some points I'm not 100% comfortable.   

- We mapped "empreendimentos" to development, which is a common name to this kind of venture but sounds too wide for me. Another common option is `land development` (this one sounds too raw for me), but it's another way to go, any thoughts on it? 

- Another point is what we call `status`. This one could be launch phase or even phase (status could be inactive/active related). Which one makes more sense for you guys?  
---
Also, I'll refactor images before merging this PR, once on our current state one image needs point to a listing, which will not be true anymore after this PR. 

Last but not least, I'll open the GraphQL queries in another PR, I guess I'll take some discovery time while doing it.

### CODE REVIEW

**Comment by pmargreff** [comment]
I guess it's good enough to merge. 

Another point to discuss, the image <-> development assoc; At first I thought about something like a polymorphic association, [exploring the possible solutions](https://hexdocs.pm/ecto/2.1.4/Ecto.Schema.html#belongs_to/3-polymorphic-associations) I opted for the simple one, probably some time we will need to migrate for something more robust and flexible, but in meantime I think it's good enough.

**Comment by pmargreff** [comment]
@rhnonose made the tests. Another point, should us validate at least one constraint (listing_id or development_id on this case) to create an image?

**Comment by pmargreff** [comment]
The build does not agree with mine affirmation, taking a look at the error.

**Comment by pmargreff** [comment]
@rhnonose build fixed, we're good to go again.

File: apps/re/lib/addresses/address.ex [hunk 1]
```diff
@@ -24,6 +24,7 @@ defmodule Re.Address do
     field :state_slug, :string
 
     has_many :listings, Re.Listing
+    has_many :developments, Re.Development
```
Comment (joaodubas) [inline apps/re/lib/addresses/address.ex]: This could also be `EstateDevelopment` or `EstateProject` (https://www.linguee.pt/portugues-ingles/traducao/empreendimento+imobiliário.html).

File: apps/re/lib/images/image.ex [hunk 2]
```diff
@@ -13,6 +13,7 @@ defmodule Re.Image do
     field :description, :string
 
     belongs_to :listing, Re.Listing
+    belongs_to :development, Re.Development
```
Comment (joaodubas) [inline apps/re/lib/images/image.ex]: In the future we could move both associations to different structs, something like:

```elixir
defmodule Re.ListingImage do
  belongs_to :image, Re.Image
  belongs_to: :listing, Re.Listing
end

defmodule Re.DevelopmentImage do
  belongs_to :image, Re.Image
  belongs_to :develoment, Re.Development
end
```

File: apps/re/lib/developments/development.ex [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  """
+
+  use Ecto.Schema
+
+  import Ecto.Changeset
+
+  schema "developments" do
+    field :name, :string
+    field :title, :string
+    field :status, :string
```
Comment (joaodubas) [inline apps/re/lib/developments/development.ex]: Maybe `phase` or `construction_status` are better suited for this. Beside that, we could have a list of possible values for this field.

File: apps/re/test/support/factory.ex [hunk 4]
```diff
@@ -143,6 +143,16 @@ defmodule Re.Factory do
 
   def tour_appointment_factory, do: %Re.Calendars.TourAppointment{}
 
+  def development_factory do
+    %{
+      name: Name.name(),
+      title: Name.name(),
+      status: Enum.random(~w(pre-launch planning building delivered)),
```
Comment (joaodubas) [inline apps/re/test/support/factory.ex]: This list should be available in `Re.Development` struct. We should set a common pattern, [`Re.Listing` use both english and portuguese for this kind of data][0].

[0]: https://github.com/emcasa/backend/blob/4d0aa97ca33e059a6735aaee39022d4707a07dde/apps/re/lib/listings/schemas/listing.ex#L55-L57

File: apps/re/lib/images/image.ex (hunk 2 above)
Comment (joaodubas) [reply apps/re/lib/images/image.ex]: Forget about this, with `Event Source` this won't be needed.

Comment (joaodubas) [inline apps/re/lib/images/image.ex]: How do we classify a given image, or this issue will be solved in another PR?

File: apps/re/lib/developments/development.ex (hunk 3 above)
Comment (pmargreff) [reply apps/re/lib/developments/development.ex]: Thank you, I'll rename it.
About the status list, it's in [line 23 in this same file](https://github.com/emcasa/backend/pull/424/files#diff-09252c9ec11d6142b678f04888d933faR23).

File: apps/re/lib/images/image.ex (hunk 2 above)
Comment (pmargreff) [reply apps/re/lib/images/image.ex]: I choose to not solve this problem yet, we'll be able to categorize images only in the future.

File: apps/re/test/support/factory.ex (hunk 4 above)
Comment (pmargreff) [reply apps/re/test/support/factory.ex]: I pointed out the status list in another comment. About the common pattern I completely agree, use English would be the best way in my opinion. Types ones which use Portuguese definitions, it would be the easiest to change.

File: apps/re/lib/addresses/address.ex (hunk 1 above)
Comment (pmargreff) [reply apps/re/lib/addresses/address.ex]: Once we're already on this context Real Estate words would be too redundant (it's my guess). Most of our benchmarks use only development` (or `development and building`), which make some sense considering the implicit context. 

Make some parallel, our listings also could be `RealStateListing`, but once we're on this context it's not necessary, this argument makes sense?

Comment (joaodubas) [reply apps/re/lib/addresses/address.ex]: Yeah, it makes sense, our product is related to real state, so we can imply that every concept belongs to real state.

File: apps/re/lib/images/image.ex (hunk 2 above)
Comment (joaodubas) [reply apps/re/lib/images/image.ex]: :ok:, this is a good approach, let's solve just one problem 👍
